test/SerialIOTest.cpp

```diff
@@ -1527,25 +1527,25 @@ TEST_CASE( "adios1_dtype_test", "[serial][adios1]" )
     REQUIRE(s.getAttribute("int32").get< int32_t >() == 32);
-    REQUIRE(getCast<int64_t>(s.getAttribute("int64")) == 64);
+    REQUIRE(s.getAttribute("int64").get<int64_t>() == 64);
     REQUIRE(s.getAttribute("uint16").get< uint16_t >() == 16u);
     REQUIRE(s.getAttribute("uint32").get< uint32_t >() == 32u);
-    REQUIRE(getCast<uint64_t>(s.getAttribute("uint64")) == 64u);
+    REQUIRE(s.getAttribute("uint64").get< uint64_t >() == 64u);
     REQUIRE(s.getAttribute("float").get< float >() == 16.e10f);
     REQUIRE(s.getAttribute("double").get< double >() == 1.e64);
-    REQUIRE(getCast< long double >(s.getAttribute("longdouble")) == 1.e80L);
+    REQUIRE(s.getAttribute("longdouble").get< long double >() == 1.e80L);
     REQUIRE(s.getAttribute("string").get< std::string >() == "string");
     REQUIRE(s.getAttribute("vecChar").get< std::vector< char > >() == std::vector< char >({'c', 'h', 'a', 'r'}));
     REQUIRE(s.getAttribute("vecInt16").get< std::vector< int16_t > >() == std::vector< int16_t >({32766, 32767}));
     REQUIRE(s.getAttribute("vecInt32").get< std::vector< int32_t > >() == std::vector< int32_t >({2147483646, 2147483647}));
-    REQUIRE(getCast< std::vector< int64_t > >(s.getAttribute("vecInt64")) == std::vector< int64_t >({9223372036854775806, 9223372036854775807}));
+    REQUIRE(s.getAttribute("vecInt64").get< std::vector< int64_t > >() == std::vector< int64_t >({9223372036854775806, 9223372036854775807}));
     REQUIRE(s.getAttribute("vecUchar").get< std::vector< char > >() == std::vector< char >({'u', 'c', 'h', 'a', 'r'}));
     REQUIRE(s.getAttribute("vecUint16").get< std::vector< uint16_t > >() == std::vector< uint16_t >({65534u, 65535u}));
     REQUIRE(s.getAttribute("vecUint32").get< std::vector< uint32_t > >() == std::vector< uint32_t >({4294967294u, 4294967295u}));
-    REQUIRE(getCast< std::vector< uint64_t > >(s.getAttribute("vecUint64")) == std::vector< uint64_t >({18446744073709551614u, 18446744073709551615u}));
+    REQUIRE(s.getAttribute("vecUint64").get< std::vector< uint64_t > >() == std::vector< uint64_t >({18446744073709551614u, 18446744073709551615u}));
     REQUIRE(s.getAttribute("vecFloat").get< std::vector< float > >() == std::vector< float >({0.f, 3.40282e+38f}));
     REQUIRE(s.getAttribute("vecDouble").get< std::vector< double > >() == std::vector< double >({0., 1.79769e+308}));
-    REQUIRE(getCast< std::vector< long double > >(s.getAttribute("vecLongdouble")) == std::vector< long double >({0.L, std::numeric_limits<long double>::max()}));
+    REQUIRE(s.getAttribute("vecLongdouble").get< std::vector< long double > >() == std::vector< long double >({0.L, std::numeric_limits<long double>::max()}));
     REQUIRE(s.getAttribute("vecString").get< std::vector< std::string > >() == std::vector< std::string >({"vector", "of", "strings"}));
     REQUIRE(s.getAttribute("bool").get< unsigned char >() == true);
     REQUIRE(s.getAttribute("boolF").get< unsigned char >() == false);
```

Comment:
This now can be retrieved as `bool`, thanks to the casting `get`.

Have this in my staged set, will push soon.

Reply:
That sounds great, thanks! :)